accelerator/core/serializer/ser_mam.c

```diff
@@ -482,3 +482,82 @@ status_t recv_mam_message_res_serialize(ta_recv_mam_res_t* const res, char** obj
   cJSON_Delete(json_root);
   return ret;
 }
+
+status_t register_mam_channel_req_deserialize(const char* const obj, ta_register_mam_channel_req_t* req) {
+  if (obj == NULL || req == NULL) {
+    ta_log_error("%s\n", ta_error_to_string(SC_SERIALIZER_NULL));
+    return SC_SERIALIZER_NULL;
+  }
+  status_t ret = SC_OK;
+  cJSON* json_obj = cJSON_Parse(obj);
+  cJSON* json_elt = NULL;
+
+  if (json_obj == NULL) {
+    ret = SC_SERIALIZER_JSON_PARSE;
+    ta_log_error("%s\n", ta_error_to_string(ret));
+    goto done;
+  }
+
+  json_elt = cJSON_GetObjectItemCaseSensitive(json_obj, "seed");
+  if (json_elt != NULL && json_elt->valuestring != NULL && strlen(json_elt->valuestring) == NUM_TRYTES_ADDRESS) {
+    strncpy(req->seed, json_elt->valuestring, NUM_TRYTES_ADDRESS);
+    req->seed[NUM_TRYTES_ADDRESS] = 0;
+  } else {
+    ret = SC_SERIALIZER_INVALID_REQ;
+    ta_log_error("%s\n", ta_error_to_string(ret));
+  }
+
+done:
+  cJSON_Delete(json_obj);
+  return ret;
+}
+
+status_t register_mam_channel_res_serialize(const char* const uuid, char** obj) {
+  if (uuid == NULL) {
+    ta_log_error("%s\n", ta_error_to_string(SC_SERIALIZER_NULL));
+    return SC_SERIALIZER_NULL;
+  }
+
+  status_t ret = SC_OK;
+  cJSON* json_root = cJSON_CreateObject();
+
+  cJSON_AddStringToObject(json_root, "user-id", uuid);
+
+  *obj = cJSON_PrintUnformatted(json_root);
+  if (*obj == NULL) {
+    ta_log_error("%s\n", ta_error_to_string(SC_SERIALIZER_JSON_PARSE));
+    ret = SC_SERIALIZER_JSON_PARSE;
+  }
+
+  cJSON_Delete(json_root);
+  return ret;
+}
+
+status_t register_mam_channel_res_deserialize(const char* const obj, char* user_id) {
+  if (obj == NULL || user_id == NULL) {
+    ta_log_error("%s\n", ta_error_to_string(SC_SERIALIZER_NULL));
+    return SC_SERIALIZER_NULL;
+  }
+  status_t ret = SC_OK;
+  cJSON* json_obj = cJSON_Parse(obj);
+  cJSON* json_elt = NULL;
+
+  if (json_obj == NULL) {
+    ret = SC_SERIALIZER_JSON_PARSE;
+    ta_log_error("%s\n", ta_error_to_string(ret));
+    goto done;
+  }
+
+  json_elt = cJSON_GetObjectItemCaseSensitive(json_obj, "user-id");
```

Comment:
Why `elf`?

Reply:
How about I changing it to `json_value`?